doc/refactoring_design.md

```diff
@@ -0,0 +1,481 @@
+# Keeper Contract Design for Tethys
+
+***DISCLAIMER: THIS IS A WORK IN PROGRESS***
+
+This document provides technical details and describes the high-level design of keeper contract in ocean protocol.
+
+## 1. Overview
+
+### 1.1 Core Modules
+
+From the top-level point of view, there are total 9 moduels as shown in the below:
+
+<img src="img/refactoring/modules.jpg" width="1000" />
+
+* **Ocean Token**: the native ERC20 token in Ocean network
+* **Exchange**: Uniswap-like exchange built inside Ocean, where user can exchange between Ether and Ocean tokens.
```

Comment:
I don't get `uniswap-like' means. Do we have our own token swapping `implementation` or Are we using `uniswap` library? which I recommend.

Reply:
we will implement the token swapping using the same mechanism of Uniswap. However, the token swapping feature is still questionable whether it should be included into Tethys. Let us decide on Thursday.